ledger/ledger-on-sql/src/main/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriter.scala

```diff
@@ -131,16 +137,17 @@ object SqlLedgerReaderWriter {
     database.inWriteTransaction("Checking ledger ID at startup") { queries =>
       val providedLedgerId =
         initialLedgerId.getOrElse(Ref.LedgerString.assertFromString(UUID.randomUUID.toString))
-      val ledgerId = queries.updateOrRetrieveLedgerId(providedLedgerId)
-      if (initialLedgerId.exists(_ != ledgerId)) {
-        Future.failed(
-          new LedgerIdMismatchException(
-            domain.LedgerId(ledgerId),
-            domain.LedgerId(initialLedgerId.get),
-          ))
-      } else {
-        Future.successful(ledgerId)
-      }
+      Future.fromTry(queries.updateOrRetrieveLedgerId(providedLedgerId).flatMap { ledgerId =>
+        if (initialLedgerId.exists(_ != ledgerId)) {
+          Failure(
+            new LedgerIdMismatchException(
+              domain.LedgerId(ledgerId),
+              domain.LedgerId(initialLedgerId.get),
+            ))
+        } else {
+          Success(ledgerId)
+        }
+      })
```

Comment:
The indentation here is a bit hard to follow. Can you make it easier to read, please? E.g., you could break the line after `Future.fromTry` or extract the construction of the `Try`.